From: Marijn Suijten <marijn.suijten@somainline.org>
Subject: [PATCH v2 02/11] dt-bindings: display: panel: Describe LGD  LH599QH3-EDB1 panel
Date: Mon, 22 Dec 2025 00:32:08 +0100

Document the LG-Display LH599QH3-EDB1 OLED DSI panel found on the Sony
Xperia XZ3.

Signed-off-by: Marijn Suijten <marijn.suijten@somainline.org>
---
 .../bindings/display/panel/lgd,lh599qh3-edb1.yaml  | 76 ++++++++++++++++++++++
 MAINTAINERS                                        |  5 ++
 2 files changed, 81 insertions(+)

diff --git a/Documentation/devicetree/bindings/display/panel/lgd,lh599qh3-edb1.yaml b/Documentation/devicetree/bindings/display/panel/lgd,lh599qh3-edb1.yaml
new file mode 100644
index 000000000000..245d14c0a1a2
--- /dev/null
+++ b/Documentation/devicetree/bindings/display/panel/lgd,lh599qh3-edb1.yaml
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/display/panel/lgd,lh599qh3-edb1.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: LG Display LH599QH3-EDB1 1440x2880 6.0" OLED DSI command-mode panel
+
+maintainers:
+  - Marijn Suijten <marijn.suijten@somainline.org>
+
+description: |
+  LG Display LH599QH3-EDB1 6.0" 1440x2880 (9:18 aspect ratio) 60Hz panel, found
+  in the Sony Xperia XZ3 smartphone.  It is always programmed with Display
+  Stream Compression 1.1 enabled.
+
+  The assembly features an Atmel maXTouch digitizer, described separately as
+  atmel,maxtouch.
+
+allOf:
+  - $ref: panel-common.yaml#
+
+properties:
+  compatible:
+    const: lgd,lh599qh3-edb1-um1
+
+  port: true
+
+  reg:
+    maxItems: 1
+    description: DSI virtual channel
+
+  reset-gpios: true
+
+  avdd-supply:
+    description: Analog voltage supply (3.0V)
+
+  vddio-supply:
+    description: I/O voltage supply (1.8V)
+
+required:
+  - compatible
+  - port
+  - reg
+  - reset-gpios
+  - avdd-supply
+  - vddio-supply
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/gpio/gpio.h>
+
+    dsi {
+        #address-cells = <1>;
+        #size-cells = <0>;
+        panel@0 {
+            compatible = "lgd,lh599qh3-edb1-um1";
+            reg = <0>;
+
+            avdd-supply = <&vreg_l28a_2p8>;
+            vddio-supply = <&vreg_l14a_1p8>;
+
+            reset-gpios = <&tlmm 6 GPIO_ACTIVE_HIGH>;
+
+            port {
+                panel_in: endpoint {
+                    remote-endpoint = <&dsi0_out>;
+                };
+            };
+        };
+    };
+
+...
+
diff --git a/MAINTAINERS b/MAINTAINERS
index 0dbf349fc1ed..6a358fee4cae 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -7931,6 +7931,11 @@ T:	git https://gitlab.freedesktop.org/drm/misc/kernel.git
 F:	Documentation/devicetree/bindings/display/panel/lg,sw43408.yaml
 F:	drivers/gpu/drm/panel/panel-lg-sw43408.c
 
+DRM DRIVER FOR LGD LH599QH3-EDB1 PANELS
+M:	Marijn Suijten <marijn.suijten@somainline.org>
+S:	Maintained
+F:	Documentation/devicetree/bindings/display/panel/lgd,lh599qh3-edb1.yaml
+
 DRM DRIVER FOR LOGICVC DISPLAY CONTROLLER
 M:	Paul Kocialkowski <paulk@sys-base.io>
 S:	Supported

-- 
2.52.0

